cmd/kube-apiserver/app/server.go

```diff
@@ -250,16 +253,41 @@ func Run(s *options.APIServer) error {
 		authorizationConfig.RBACClusterRoleBindingRegistry = clusterrolebinding.NewRegistry(clusterrolebindingetcd.NewREST(mustGetRESTOptions("clusterrolebindings")))
 	}
 
-	authorizer, err := authorizer.NewAuthorizerFromAuthorizationConfig(authorizationModeNames, authorizationConfig)
+	apiAuthorizer, err := authorizer.NewAuthorizerFromAuthorizationConfig(authorizationModeNames, authorizationConfig)
 	if err != nil {
 		glog.Fatalf("Invalid Authorization Config: %v", err)
 	}
 
 	admissionControlPluginNames := strings.Split(s.AdmissionControl, ",")
-	client, err := s.NewSelfClient()
+
+	var token string
+	var secureOnly = s.InsecurePort == 0
+	if secureOnly {
+		token = uuid.NewRandom().String()
+	}
+
+	client, err := s.NewSelfClient(secureOnly, token)
 	if err != nil {
 		glog.Errorf("Failed to create clientset: %v", err)
 	}
+
+	if secureOnly {
+		// Set up a loop back authenticator and authorizer since insecure port is turned off
+		var uid = uuid.NewRandom().String()
+		tokenAuthenticator, err := authenticator.NewAuthenticatorFromToken(token, "system:apiserver", uid, "")
```

Comment:
How about `user == "system:apiserver"` and `groups = []string{"system:masters"}` and we do authz by the group.  That will make it more reasonable easier to mint privileged certificates for infrastructure components with unique names that identify them.